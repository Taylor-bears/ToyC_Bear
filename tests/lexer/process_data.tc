int process_data_1(int input_val) {
    if (input_val >= 100 && input_val != 200) {
        return (input_val % 10) * -1; 
    } else {
        return 0;
    }
}