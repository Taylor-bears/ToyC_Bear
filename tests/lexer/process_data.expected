INT
ID(process_data_1)
LPAREN
INT
ID(input_val)
RPAREN
LBRACE
IF
LPAREN
ID(input_val)
GE
NUMBER(100)
AND
ID(input_val)
NEQ
NUMBER(200)
RPAREN
LBRACE
RETURN
LPAREN
ID(input_val)
MOD
NUMBER(10)
RPAREN
TIMES
MINUS
NUMBER(1)
SEMICOLON
RBRACE
ELSE
LBRACE
RETURN
NUMBER(0)
SEMICOLON
RBRACE
RBRACE
EOF